CMakeLists.txt

```diff
@@ -46,73 +46,67 @@ target_include_directories(
   triton-common-json
   INTERFACE
     $<INSTALL_INTERFACE:include>
-    $<INSTALL_INTERFACE:${RapidJSON_INCLUDE_DIRS}>
+    $<INSTALL_INTERFACE:${RAPIDJSON_INCLUDE_DIRS}>
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-    $<BUILD_INTERFACE:${RapidJSON_INCLUDE_DIRS}>
+    $<BUILD_INTERFACE:${RAPIDJSON_INCLUDE_DIRS}>
 )
 
 #
 # Table Printer
 #
-if(NOT WIN32)
-  add_library(
-    triton-common-table-printer
-    src/table_printer.cc
-  )
-  
-  add_library(
-    TritonBackend::triton-common-table-printer ALIAS triton-common-table-printer
-  )
-  
-  target_include_directories(
-    triton-common-table-printer
-    PUBLIC
-      $<INSTALL_INTERFACE:include>
-      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-    PRIVATE
-      ${CMAKE_CURRENT_SOURCE_DIR}/src
-  )
-  
-  target_compile_features(triton-common-table-printer PRIVATE cxx_std_11)
-  target_compile_options(
-    triton-common-table-printer
-    PRIVATE
-      -Wall -Wextra -Wno-unused-parameter -Werror
-  )
-  
-  set_target_properties(
-    triton-common-table-printer PROPERTIES
-    POSITION_INDEPENDENT_CODE ON
-    OUTPUT_NAME tritontableprinter
-  )
-endif() # NOT WIN32
+add_library(
+  triton-common-table-printer
+  src/table_printer.cc
+)
+
+add_library(
+  TritonBackend::triton-common-table-printer ALIAS triton-common-table-printer
+)
+
+target_include_directories(
+  triton-common-table-printer
+  PUBLIC
+    $<INSTALL_INTERFACE:include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  PRIVATE
+    ${CMAKE_CURRENT_SOURCE_DIR}/src
+)
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  message("Using MSVC as compiler, default target on Windows 10. "
+		  "If the target system is not Windows 10, please update _WIN32_WINNT "
+		  "to corresponding value.")
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
```

Comment:
use
set_target_properties(triton-common-table-printer PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
Perhaps can just add it to the set_target_properties below.